### PR TITLE
bpo-39852: IDLE 'Go to line' deletes selection, updates status

### PR DESCRIPTION
**Pull request by terryjreedy**
Clearing any selection prevents accidental deletion.
The update is for Ln and Col on the status bar.

https://bugs.python.org/issue39852

### CODE REVIEW

**Comment by terryjreedy** [comment]
Tal and Cheryl: it would be nice to have this tested on Mac and Linux (Ubuntu).  A minimal test: in shell, enter something after the prompt and select some of it.  Open goto box, enter 2, *close by clicking OK*  Selected text should remain but selection highlight should be gone.  Status bar should be updated.  Previously, at least on windows, selection highlight remained and status Ln, Col were not updated.  For more: Repeat but close with Enter key, and without selection. Everything should work in editor too.

**Comment by terryjreedy** [comment]
The patch to idlelib/help.html included changes to idle.rst in #18771 that were not propagated.

File: Lib/idlelib/editor.py [hunk 1]
```diff
@@ -678,8 +678,13 @@ def goto_line_event(self, event):
         if lineno <= 0:
             text.bell()
             return "break"
-        text.mark_set("insert", "%d.0" % lineno)
+        newindex = f'{lineno}.0'
+        text.mark_set("insert", newindex )
```
Comment (csabella) [inline Lib/idlelib/editor.py]: Rogue extra space after `newindex`.

**Comment by csabella** [comment]
There's a `remove_selection` method in `editor.py`.  I looked at what you had written, but it seems that using `remove_selection` (with or without `selection_clear()`) fixes the issue.  Since you didn't mention calling `remove_selection`, I was wondering if you tried it?

File: Lib/idlelib/editor.py (hunk 1 above)
Comment (terryjreedy) [reply Lib/idlelib/editor.py]: In response to next comment, I reverted mark_set change.

**Comment by terryjreedy** [comment]
I was not currently aware of it.  Since I want .see after .mark_set, I pulled in .tag_remove, which for some reason I did not think was working.  Bot it does here.  Thank you.

**Review by csabella — APPROVED** [review]
LGTM.  This works on Ubuntu.

**Comment by terryjreedy** [comment]
I should remember to use Module Browser to scan the 100+ EditorWindow methods.

**Pull request merged**